node/ChainwebNode.hs

```diff
@@ -299,20 +305,32 @@ node conf logger = do
     dbBaseDir <- getDbBaseDir conf
     when (_nodeConfigResetChainDbs conf) $ removeDirectoryRecursive dbBaseDir
     rocksDbDir <- getRocksDbDir conf
+    rocksDbCheckpointDir <- getRocksDbCheckpointDir conf
     pactDbDir <- getPactDbDir conf
     withRocksDb rocksDbDir $ \rocksDb -> do
         logFunctionText logger Info $ "opened rocksdb in directory " <> sshow rocksDbDir
-        withChainweb cwConf logger rocksDb pactDbDir (_nodeConfigResetChainDbs conf) $ \cw -> mapConcurrently_ id
-            [ runChainweb cw
-              -- we should probably push 'onReady' deeper here but this should be ok
-            , runCutMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
-            , runQueueMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
-            , runRtsMonitor (_chainwebLogger cw)
-            , runBlockUpdateMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
-            ]
+        bracket (newIORef (Just rocksDb)) (flip writeIORef Nothing) $ \ref -> do
```

Comment:
This is extremely paranoid, perhaps too paranoid, about not letting us access a `RocksDb` when it's out of scope.

Reply:
Specifically, this is for not letting the signal handler access a `RocksDb` when it's out of scope, but that assumes that we can exit this `bracket` *and then* trigger the signal handler without the program having terminated.